tests/scancode/test_cli.py

```diff
@@ -463,3 +464,17 @@ def test_scan_can_run_from_other_directory():
         print(stderr)
     assert rc == 0
     check_scan(test_env.get_test_loc(expected_file), result_file, strip_dates=True, regen=False)
+
+
+class TestCLI(TestCase):
+
+	def test_shorten_filename(self):
+	    assert '0123456789...0123456789.c' == cli.shorten_filename('0123456789012345678901234567890123456789.c')
+	    assert '012345.c' == cli.shorten_filename('012345.c')
+	    assert '012345678901234567890.cpp' == cli.shorten_filename('012345678901234567890.cpp')
```

Comment:
you should add tests for limit args: empty, None, list, string of a length that is and a length that is not shortened right around your 25 limit.  It may help to use variables and give names to make things more explicit. Also you need to test things without extensions and full paths.
In general it is better to create one test function for one test, though the example I gave you did not follow that pattern.